session.go

```diff
@@ -1,97 +1,77 @@
 package bond
 
 import (
+	"database/sql"
+	"fmt"
 	"reflect"
 	"sync"
 
-	"upper.io/db"
+	"upper.io/db.v2"
+	"upper.io/db.v2/lib"
 )
 
 type Session interface {
-	db.Tx
+	lib.SQLDatabase
```

Comment:
first idea, how come we make bond use SQLDatabase.. perhaps instead we could have a TxDatabase instead...? which is more of a generic interface to a database with transactions. and of course, some db's might not support it. 

second idea, perhaps db.Database should have the TxDatabase methods in it by default..? then for databases that dont support it, they return an error saying unsupported, it's not so bad that way, and fewer interfaces, and most db's do have transactions of some kind